### PR TITLE
[ DOC ] setBatchSize update

### PR DESCRIPTION
**Pull request by jane-intel**

### CODE REVIEW

File: docs/IE_DG/ShapeInference.md [hunk 1]
```diff
@@ -27,29 +27,20 @@ OpenVINO™ provides the following methods for runtime model reshaping:
...
-               -> reshape(input_2_shape) -> LoadNetwork -> infer(input_2)
-   ```
    > **NOTES**:
    > - Starting with the 2021.1 release, the Model Optimizer converts topologies keeping shape-calculating sub-graphs by default, which enables correct shape propagation during reshaping.
    > - Older versions of IRs are not guaranteed to reshape successfully. Please regenerate them with the Model Optimizer of the latest version of OpenVINO™.<br>
    > - If an ONNX model does not have a fully defined input shape and the model was imported with the ONNX importer, reshape the model before loading it to the plugin.
 * **Set a new batch dimension value** with the `InferenceEngine::CNNNetwork::setBatchSize` method.<br>     
    The meaning of a model batch may vary depending on the model design.
-   The `InferenceEngine::CNNNetwork::setBatchSize` method deduces the index of a batch dimension based only on the input rank. 
+   Batch dimension assumed to be placed on the 0 index for all inputs of the model.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:36]: --> `Batch dimension is usually placed at the 0 index of all inputs of the model. `

File: docs/IE_DG/ShapeInference.md [hunk 2]
```diff
@@ -27,29 +27,20 @@ OpenVINO™ provides the following methods for runtime model reshaping:
...
-   The batch-setting algorithm does not involve the shape inference mechanism.
-   Batch of input and output shapes for all layers is set to a new batch value without layer validation.
-   It may cause both positive and negative side effects.
-   Due to the limitations described above, the current method is not recommended to use.
-   If you need to set a new batch size for the model, use the `CNNNetwork::reshape` method instead.
-
-Do not use runtime reshaping methods simultaneously, especially do not call the `CNNNetwork::reshape` method after you use `InferenceEngine::CNNNetwork::setBatchSize`.
-The `InferenceEngine::CNNNetwork::setBatchSize` method causes irreversible conversion of the internal model representation into the legacy model representation.
-The method does not use nGraph for shape inference which leads to reduced reshape opportunities and may affect the performance of the model.
+   The `InferenceEngine::CNNNetwork::setBatchSize` method transforms the model before new shape propagation to relax hard-coded batch dimension in the model if any.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:38]: --> `The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.`

File: docs/IE_DG/ShapeInference.md [hunk 3]
```diff
@@ -27,29 +27,20 @@ OpenVINO™ provides the following methods for runtime model reshaping:
...
+
+You can change input shapes of a model multiple times using `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:40]: --> 
```To change input shapes multiple times, use the `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.```

File: docs/IE_DG/ShapeInference.md [hunk 4]
```diff
@@ -27,29 +27,20 @@ OpenVINO™ provides the following methods for runtime model reshaping:
...
+If a model has hard-coded batch dimension, use `InferenceEngine::CNNNetwork::setBatchSize` first to change the batch and then call `InferenceEngine::CNNNetwork::reshape` to update other dimensions if needed.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:41]: --> ```If a model has a hard-coded batch dimension, use `InferenceEngine::CNNNetwork::setBatchSize` first to change the batch, then call `InferenceEngine::CNNNetwork::reshape` to update other dimensions, if needed.```

File: docs/IE_DG/ShapeInference.md [hunk 5]
```diff
@@ -27,29 +27,20 @@ OpenVINO™ provides the following methods for runtime model reshaping:
...
+
+Once input shape of the `InferenceEngine::CNNNetwork` is set call `InferenceEngine::Core::LoadNetwork` method to get `InferenceEngine::ExecutableNetwork` object for inference with updated shapes.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:43]: --> ```Once the input shape of `InferenceEngine::CNNNetwork` is set, call the `InferenceEngine::Core::LoadNetwork` method to get an `InferenceEngine::ExecutableNetwork` object for inference with updated shapes.```

**Review by aalborov — CHANGES_REQUESTED** [review]

File: docs/IE_DG/ShapeInference.md [hunk 6]
```diff
@@ -62,8 +53,8 @@ Shape collision during shape propagation may be a sign that a new shape does not
 Changing the model input shape may result in intermediate operations shape collision.
 
 Examples of such operations:
-- <a href="_docs_MO_DG_prepare_model_convert_model_IR_V10_opset1.html#Reshape">`Reshape` operation</a> with a hard-coded output shape value
-- <a href="_docs_MO_DG_prepare_model_convert_model_IR_V10_opset1.html#MatMul">`MatMul` operation</a> with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
+- <a href="_docs_ops_shape_Reshape_1.html">`Reshape` operation</a> with a hard-coded output shape value
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:56]: These should work:
```
- [`Reshape` operation](../ops/shape/Reshape_1.md) with a hard-coded output shape value
- [`MatMul` operation](../ops/matrix/MatMul_1.md) with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```

File: docs/IE_DG/ShapeInference.md [hunk 7]
```diff
@@ -62,8 +53,8 @@ Shape collision during shape propagation may be a sign that a new shape does not
...
+- <a href="_docs_ops_matrix_MatMul_1.html">`MatMul` operation</a> with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```
Comment (jane-intel) [inline docs/IE_DG/ShapeInference.md:57]:
```suggestion
- [`Reshape` operation](../ops/shape/Reshape_1.md) with a hard-coded output shape value
- [`MatMul` operation](../ops/matrix/MatMul_1.md) with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```

Comment (jane-intel) [inline docs/IE_DG/ShapeInference.md:57]:
```suggestion
- [`Reshape` operation](../ops/shape/Reshape_1.md) with a hard-coded output shape value
- [`MatMul` operation](../ops/matrix/MatMul_1.md) with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
```

**Review by aalborov — APPROVED** [review]

File: docs/IE_DG/ShapeInference.md [hunk 8]
```diff
@@ -27,29 +27,22 @@ OpenVINO™ provides the following methods for runtime model reshaping:
...
+   Batch dimension is usually placed at the 0 index of all inputs of the model.
    This method does not work for models with a non-zero index batch placement or models with inputs without a batch dimension. 
-   The batch-setting algorithm does not involve the shape inference mechanism.
-   Batch of input and output shapes for all layers is set to a new batch value without layer validation.
-   It may cause both positive and negative side effects.
-   Due to the limitations described above, the current method is not recommended to use.
-   If you need to set a new batch size for the model, use the `CNNNetwork::reshape` method instead.
+   The method transforms the model before a new shape propagation to relax a hard-coded batch dimension in the model, if any.
+
+To change input shapes multiple times, use the `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.
```
Comment (aalborov) [inline docs/IE_DG/ShapeInference.md:42]:
```suggestion
You can change input shapes multiple times using the `InferenceEngine::CNNNetwork::reshape` and `InferenceEngine::CNNNetwork::setBatchSize` methods in any order.
```

**Pull request merged**